Source/WebCore/page/SecurityOriginData.h

```diff
@@ -110,52 +157,19 @@ struct SecurityOriginData {
 #endif
 
     static bool shouldTreatAsOpaqueOrigin(const URL&);
-
-    template<class Encoder> void encode(Encoder&) const;
-    template<class Decoder> static std::optional<SecurityOriginData> decode(Decoder&);
 };
 
-template<class Encoder>
-void SecurityOriginData::encode(Encoder& encoder) const
-{
-    encoder << opaqueOriginIdentifier;
-    if (opaqueOriginIdentifier)
-        return;
-    encoder << protocol << host << port;
-}
-
-template<class Decoder>
-std::optional<SecurityOriginData> SecurityOriginData::decode(Decoder& decoder)
-{
-    std::optional<SecurityOriginData::MarkableOpaqueOriginIdentifier> opaqueOriginIdentifier;
-    decoder >> opaqueOriginIdentifier;
-    if (!opaqueOriginIdentifier)
-        return std::nullopt;
-    if (*opaqueOriginIdentifier)
-        return SecurityOriginData(**opaqueOriginIdentifier);
-    std::optional<String> protocol;
-    decoder >> protocol;
-    if (!protocol)
-        return std::nullopt;
-    if (protocol->isHashTableDeletedValue())
-        return std::nullopt;
-    std::optional<String> host;
-    decoder >> host;
-    if (!host)
-        return std::nullopt;
-    std::optional<std::optional<uint16_t>> port;
-    decoder >> port;
-    if (!port)
-        return std::nullopt;
-    return SecurityOriginData(WTFMove(*protocol), WTFMove(*host), *port);
-}
-
 WEBCORE_EXPORT bool operator==(const SecurityOriginData&, const SecurityOriginData&);
 inline bool operator!=(const SecurityOriginData& first, const SecurityOriginData& second) { return !(first == second); }
 
 inline void add(Hasher& hasher, const SecurityOriginData& data)
 {
-    add(hasher, data.protocol, data.host, data.port, data.opaqueOriginIdentifier);
+    add(hasher, data.data.index());
```

Comment:
we already have a `add(Hasher&)` overload for `std::variant` so let's not duplicate the logic here.
We should add a `add(Hasher&)` overload for `SecurityOriginData::tuple` and then just call `add(hasher, data)` here.